Src/FluentAssertions/Specialized/AsyncFunctionAssertions.cs

```diff
@@ -10,37 +9,20 @@ namespace FluentAssertions.Specialized
     /// Contains a number of methods to assert that an asynchronous method yields the expected result.
     /// </summary>
     [DebuggerNonUserCode]
-    public class AsyncFunctionAssertions
+    public class AsyncFunctionAssertions : ActionAssertions
     {
-        private readonly IExtractExceptions extractor;
-
-        public AsyncFunctionAssertions(Func<Task> subject, IExtractExceptions extractor)
+        public AsyncFunctionAssertions(Func<Task> subject, IExtractExceptions extractor) : base(() =>
+        {
+            subject().GetAwaiter().GetResult();
+        }, extractor)
         {
-            this.extractor = extractor;
             Subject = subject;
         }
 
         /// <summary>
         /// Gets the <see cref="Func{Task}"/> that is being asserted.
         /// </summary>
-        public Func<Task> Subject { get; private set; }
-
-        /// <summary>
-        /// Asserts that the current <see cref="Func{Task}"/> throws an exception of type <typeparamref name="TException"/>.
```

Comment:
❓ This changes the xml summary from `Func{Task}` to `Action` in `ActionAssertions`.

Reply:
This is a good point. I guess we have a few solutions: 

1. We could add methods in AsyncFunctionAssertions which have the same signatures and just call the base methods, but with their own XML summaries. Downside to this would be having method signatures for the sake of the comments, which is fine but might be a bit ugly.
Another downside is that anyone adding functionality to ActionAssertions will need to remember to go to any subclasses and add a new method stub with the dedicated summary. 

2. We could remove all mentions of either `Action` or `Task<Func>` in the summaries and just refer to the `Subject` property. Might look like this:

```csharp
/// <summary>
/// Asserts that the current Subject throws an exception of type <typeparamref name="TException"/>.
```
Downside to this is that "Subject" doesn't mean much to the developer. 

3. We could change the summary to show multiple types, so it might look something look:
```csharp
/// <summary>
/// Asserts that the current <see cref="Action"/> or <see cref="Func{Task}"/> throws an exception of type <typeparamref name="TException"/>.
```
Although a downside to this is that ActionAssertions is therefore then aware of AsyncFunctionAssertions, which it probably shouldn't be. 

4. We could ignore it. Using a `Func<Task>` is obviously a different type to `Action`, but the developer might understand that methods like Throw, NotThrow, etc. are synchronous, thereby essentially making their asynchronous action a regular Action. 

@dennisdoomen  @jnyrup Which solution would you guys prefer? Any other solutions I haven't thought of? If need be then we can discuss different approaches to this refactoring altogether.

Reply:
@dennisdoomen @jnyrup  I can also replace the text so it refers to it as a "delegate". I mean both Action and Func<Task> are just delegates anyway. 

```csharp
/// <summary>
/// Asserts that the current delegate throws an exception of type <typeparamref name="TException"/>.
```

What are your thoughts on what to do about the comments?

Reply:
I dislike 1 and 2 for the same reasons as you.

Referring simply to delegates would be fine with me.
That seems to be in line with `ReferenceTypes` which uses the similar general phrase `Asserts that the object`.

Reply:
Perfect, thanks very much for the feedback 👍

Reply:
@jnyrup I pushed the latest changes with "delegate". Resolved this conversation for now.